modules/apps/transfer/module.go

```diff
@@ -213,6 +214,10 @@ func ValidateTransferChannelParams(
 		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "invalid port: %s, expected %s", portID, boundPort)
 	}
 
+	if portID != counterparty.PortId {
+		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "portID: %s does not match counterparty portID: %s", portID, counterparty.PortId)
+	}
```

Comment:
Hmm, why is this check needed? Applications like interchain accounts have portIDs which do not match counterparty portIDs

Reply:
Sure but it's true for transfer. Note this is the transfer module not middleware